rfcs/62-editor-data-model.md

```diff
@@ -0,0 +1,562 @@
+# Feature Name: `editor_data_model`
+
+## Summary
+
+This RFC describes the data model and API that the Editor uses in memory, and the basis for communicating with the Game process and for serializing that data to disk (although the serialization format to disk is outside of the scope of this RFC). The data model is largely inspired by the one of [Our Machinery](https://ourmachinery.com/), as described in [this blog post](https://ourmachinery.com/post/the-story-behind-the-truth-designing-a-data-model/).
+
+## Glossary
+
+In this RFC, the following terms are used:
+
+- **Game** refers to any application made with Bevy, which encapsulates proper games but also any other kind of Bevy application (CAD, architecture software, _etc._).
+- **Author** refers to any person (including non-technical) building the Game with the intent of shipping it to end users.
+- **Editor** refers to the software used to edit Game data and prepare them for runtime consumption, and which this RFC describes the data model of.
+- **Runtime** refers to the Game or Editor process running time, as opposed to compiling or building time. When not otherwise specified, this refers to the Game's runtime. In general it's synonymouns of a process running, and is opposed to _build time_ or _compile time_.
+- **Project** refers to the Editor's representation of the Game that data is being created/edited for, and all the assets, metadata, settings, or any other data the Editor needs and which is associated to that specific Game. There's a 1:1 mapping between a Game and an Editor project.

```

Comment:
> There's a 1:1 mapping between a Game and an Editor project.

I think that's a very risky assumption. Think for a moment about source to machine code translation. Comments in the source do not alter the compiled program, so it's a `n:1` mapping. Similarly, I think there could be some Project data that won't affect the Game in any way, e.g. some kind of project metadata, like the project name itself, that could be different from say, the Game's window title.

Reply:
Sorry that was not about the data, I simply meant each project corresponds to a different Game, and each Game is represented in the Editor by one project. That's all.

Reply:
I'm deleting that sentence, I think it's clear enough without it, and will prevent any confusion.

Reply:
Done.